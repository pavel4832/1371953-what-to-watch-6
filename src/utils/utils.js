import {FILTER_TYPE} from '../const';

export const getTimeFromMins = (mins) => {
  let hours = Math.trunc(mins / 60);
  let minutes = mins % 60;

  return hours + `h ` + minutes + `m`;
};

export const getMovieById = (movies, id) => {
  return movies.find((movie) => movie.id === id);
};

export const getMoviesByGenre = (movies, genre) => {
  if (genre === FILTER_TYPE.ALL_GENRE) {
    return movies;
  }
  return movies.filter((movie) => movie.genre === genre);
};

const getNumber = (number) => {
  if (number.toString().length === 1) {
    return number.toString().padStart(2, `0`);
  }

  return number.toString();
};

export const formatTime = (time, hours) => {
  if (hours) {
    const h = getNumber(Math.floor(time / 3600));
    time = time - h * 3600;

    const m = getNumber(Math.floor(time / 60));
    const s = getNumber(Math.floor(time % 60));

    return h + `:` + m + `:` + s;
  } else {
    const m = getNumber(Math.floor(time / 60));
    const s = getNumber(Math.floor(time % 60));

    return m + `:` + s;
  }
};
