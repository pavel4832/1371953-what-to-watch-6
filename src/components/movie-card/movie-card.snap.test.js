import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import MovieCard from './movie-card';
import {Provider} from "react-redux";
import MOVIE from '../../mock/movie';
import {adaptMoviesToApp} from '../../utils/adaptor';

const mockStore = configureStore({});

const mockDispatch = jest.fn();
jest.mock(`react-redux`, () => ({
  ...jest.requireActual(`react-redux`),
  useDispatch: () => mockDispatch,
}));

jest.spyOn(redux, `useSelector`);

it(`Should MovieCard render correctly`, () => {
  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={mockStore({})}>
        <Router history={history}>
          <MovieCard
            movie={adaptMoviesToApp(MOVIE)}
            renderPlayer={jest.fn()}
          />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
