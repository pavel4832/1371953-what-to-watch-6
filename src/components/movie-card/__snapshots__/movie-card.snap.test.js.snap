// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MovieCard render correctly 1`] = `
<div>
  <article
    class="small-movie-card catalog__movies-card"
  >
    <div
      class="small-movie-card__image"
    />
    <h3
      class="small-movie-card__title"
    >
      <a
        class="small-movie-card__link"
        href="/films/1"
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
</div>
`;
